pkgs/misc/ghostscript/default.nix

```diff
@@ -139,7 +145,23 @@ stdenv.mkDerivation rec {
 
   # dynamic library name only contains maj.min, eg. '9.53'
   dylib_version = lib.versions.majorMinor version;
-  preFixup = lib.optionalString stdenv.isDarwin ''
+  preFixup = ''
+    for bin in $out/bin/{eps2eps,gsbj,gsdj,gsdj500,gslj,gslp,gsnd,printafm,pphs,pdf2ps} $out/bin/pf* ; do
+      wrapProgram $bin --prefix PATH : ${lib.makeBinPath [ coreutils ]}
+    done
+
+    for bin in $out/bin/ps2* ; do
+      wrapProgram $bin --prefix PATH : ${lib.makeBinPath [ coreutils ]}:$out/bin
```

Comment:
Have you checked that `$0` is interpreted correctly? I believe `wrapProgram` breaks `$0` for non-binary executables. I suspect in this case it would only break the help output, showing e.g. `.ps2pdf-wrapped` instead of `ps2pdf`. But `$0` is also used to find the sibling executables.

(By the way, for many of these scripts, a single `substituteInPlace` of the entire block `GS_EXECUTABLE= ... GS_EXECUTABLE="$gs"` with `GS_EXECUTABLE='${out}/bin/gs'` is enough, and possibly more appropriate than a wrapper.)

Reply:
Yes, help output is showing `.ps2pdf-wrapped` since it is using `basename $0`.

Substituting `GS_EXECUTABLE='${out}/bin/gs'` seems not enough because we also need to bind some PATH from coreutils, gawk.

Reply:
> Substituting `GS_EXECUTABLE='${out}/bin/gs'` seems not enough because we also need to bind some PATH from coreutils, gawk.

I meant, for quite a few scripts, the only coreutils usage is around `GS_EXECUTABLE`, so if you patch that away, there is no need for a wrapper. Other scripts do use gawk, basename, etc. so I agree with wrapping for those of course.